include/openthread/thread.h

```diff
@@ -869,6 +869,41 @@ void otThreadSetDiscoveryRequestCallback(otInstance *                     aInsta
                                          otThreadDiscoveryRequestCallback aCallback,
                                          void *                           aContext);
 
+/**
+ * This function pointer type defines the callback to notify the outcome of a `otThreadLocateAnycastDestination()`
+ * request.
+ *
+ * @param[in] aContext            A pointer to an arbitrary context (provided when callback is registered).
+ * @param[in] aMeshLocalAddress   A pointer to the mesh-local EID of the closest destination of the anycast address.
+ *                                When no destination can be found (or in case of any failure), it will be NULL.
+ * @param[in] aRloc16             The RLOC16 of the destination if found, otherwise invalid RLOC16 (0xfffe).
+ *
+ *
+ */
+typedef void (*otThreadAnycastLocatorCallback)(void *aContext, const otIp6Address *aMeshLocalAddress, uint16_t aRloc16);
```

Comment:
Is it meaningful to allow returning other addresses (e.g. DUA address is more reliable than ML-EID) ?

Reply:
I think mesh-local unicast address is good for this puporse. Anycast address itself is also mesh-local scope.